src/Microsoft.DocAsCode.Build.ReferenceBase/ReferenceDocumentProcessorBase.cs

```diff
@@ -0,0 +1,52 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.DocAsCode.Build.ReferenceBase
+{
+    using System;
+    using System.Collections.Immutable;
+
+    using Microsoft.DocAsCode.Build.Common;
+    using Microsoft.DocAsCode.Plugins;
+
+    /// <summary>
+    /// Base document processor for reference.
+    /// </summary>
+    public abstract class ReferenceDocumentProcessorBase : DisposableDocumentProcessor
```

Comment:
Will extract some logic to this base class.

---
In reply to: [93552331](https://github.com/dotnet/docfx/pull/1030#discussion_r93552331) [](ancestors = 93552331)